File: Kotlin/fraction.kt
class Fraction(var numerator: Int, var denominator: Int) {
    init {
        val gcd = _gcd(numerator, denominator)
        numerator = numerator/gcd
        denominator = denominator/gcd
    }    

    fun _gcd(a: Int, b: Int) : Int {
        if (a == b) {
            return a
        } else {
            if (a > b) {
                return _gcd(a-b, b)
            } else {
                return _gcd(a, b-a)
            }
        }
    }

    infix fun plus(f: Fraction): Fraction {
        return Fraction(this.numerator*f.denominator+this.denominator*f.numerator, this.denominator*f.denominator)
    }

    infix fun mult(f: Fraction): Fraction {
        return Fraction(this.numerator*f.numerator, this.denominator*f.denominator)
    }

    override fun toString() : String {
        return "${this.numerator} / ${this.denominator}"
    }
}

fun main(args: Array<String>) {
    var half: Fraction = Fraction(1, 2)
    var onethird: Fraction = Fraction(1, 3)
    var twothird: Fraction = Fraction(2, 3)

    var f1 = half plus onethird
    println("f1 = $f1")
    
    var f2 = half plus twothird
    println("f2 = $f2")

    var f3: Fraction = Fraction(6, 76)
    println("6 / 76 = $f3")

    var f4 = half mult twothird
    println("f4 = $f4")
}

File: Go/fraction.go
package main

import (
	"fmt"
)

type Fraction struct {
	numerator int
	denominator int
}

func _gcd(a int, b int) int {
	if (a == b) {
		return a
	} else {
		if (a > b) {
			return _gcd(a - b, b)
		} else {
			return _gcd(a, b - a)
		}
	}
}

func (f1 Fraction) add(f2 Fraction) Fraction {
	var n, d, gcd int
	var f Fraction

	n = f1.denominator * f2.numerator + f1.numerator * f2.denominator
	d = f1.denominator * f2.denominator
	gcd = _gcd(n, d)

	f.numerator = n / gcd
	f.denominator = d / gcd

	return f
}


func main() {
	f1 := Fraction{numerator: 1, denominator: 3}
	f2 := Fraction{numerator: 2, denominator: 5}
	f3 := Fraction{numerator: 2, denominator: 4}

	var f4 = f1.add(f2)
	var f5 = f1.add(f3)

	fmt.Printf("%d / %d\n", f4.numerator, f4.denominator)
	fmt.Printf("%d / %d\n", f5.numerator, f5.denominator)
}
